src/lib.rs

```diff
@@ -130,36 +130,24 @@ use doc_comment::doctest;
 doctest!("../Readme.md");
 
 // Check that the type of the expression passed here implements IntoIterator.
-// Hopefully this optimizes away in release builds.
 #[doc(hidden)]
-#[macro_export]
-macro_rules! assert_impl_into_iter {
-    ($x: expr) => {
-        let _ = || {
-            fn assert_impl_into_iter<T>(_: T)
-            where
-                T: IntoIterator,
-            {
-            }
-            assert_impl_into_iter($x);
-        };
-    };
-}
+#[inline(always)]
+pub fn assert_impl_into_iter<T: IntoIterator + ?Sized>(_: &T) {}
```

Comment:
after looking at [Edition Guide - Macro changes](https://doc.rust-lang.org/edition-guide/rust-2018/macros/macro-changes.html) again, looks like they suggest naming `#[doc(hidden)] pub ...` items that may be accidentally imported by users with the patters `__<crate-name>_<item-name>`. This is probably beyond the scope of this PR but if you're changing this definition, do you think it's a good idea to also rename it in the style that they suggest?

Reply:
This isn't a macro.

Reply:
True, but its still an item that may be imported into a users scope if they do `use py_comp::*`. The pattern they suggest reduces the risk of name collisions.

(Does rust have a way to specify what gets imported when using the `use foo::*` syntax, like Pythons `__all__` variable?

Reply:
I don't see any good reason for users to `use py_comp::*`, honestly.

Reply:
They don't, but nothing is stopping them from doing so, in which case they'll get unexpected behavior. I just want to avoid someone opening an issue about that in the future (as unlikely as that might be)